modules/300-prometheus/templates/prometheus/prometheus.yaml

```diff
@@ -176,10 +176,14 @@ spec:
 {{- $alertmanagers_byservice := false }}
 {{- $alertmanagers_internal := false }}
 {{- if hasKey .Values.prometheus.internal.alertmanagers "byService" }}
-  {{- $alertmanagers_byservice = true }}
+  {{- if gt (len .Values.prometheus.internal.alertmanagers.byService) 0 }}
```

Comment:
NIT: `{{ if len .Values.prometheus.internal.alertmanagers.byService }}` should be enough.

Reply:
sounds good, i add this to https://github.com/deckhouse/deckhouse/pull/1785/files#diff-7325ffa471bdaa881771304df9a4eda18ec949514b34c7f59fe8d29b28efa2c7